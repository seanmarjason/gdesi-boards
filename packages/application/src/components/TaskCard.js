import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import Link from '@mui/material/Link';

import { usePathname } from 'next/navigation'

import { Draggable } from '@hello-pangea/dnd';

export const TaskCard = ({card, index}) => {
    const pathname = usePathname()

    const handleTaskClick = (event) => {
        // event.preventDefault()
    }

    return(
        <Draggable
            draggableId={`${card.id}`}
            index={index}
        >
            {(provided, snapshot) => 
                <Card sx={{ minWidth: 275 }} ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
                            <CardContent>
                                <Typography gutterBottom sx={{ fontSize: 14 }}>
                                    <Link href={`${pathname}/tasks/${card.id}`} underline="none" onClick={handleTaskClick}>
                                        {card.id}
                                    </Link>
                                </Typography>
                                <Typography variant="h5" component="div">
                                {card.title}
                                </Typography>
                                <Typography sx={{ color: 'text.secondary', mb: 1.5 }}>
                                {card.type}
                                </Typography>
                                <Typography variant="body2">
                                {card.assignee}
                                <br />
                                </Typography>
                            </CardContent>
                </Card>
            }
        </Draggable>
    )
}