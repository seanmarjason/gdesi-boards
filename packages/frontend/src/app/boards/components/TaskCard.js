import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import CardActionArea from '@mui/material/CardActionArea';
import Typography from '@mui/material/Typography';

import { Draggable } from '@hello-pangea/dnd';

export const TaskCard = ({innerRef, id, title, type, assignee, provided}) => {
    return(
        // <Draggable
        //     draggableId={`draggable-${index}`}
        //     index={index}
        // >
        // {(provided, snapshot) => 
            // <div
            //     ref={provided.innerRef}
            // >
            <Card sx={{ minWidth: 275 }} ref={innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
                <CardActionArea>
                        <CardContent>
                            <Typography gutterBottom sx={{ color: 'text.secondary', fontSize: 14 }}>
                            # {id}
                            </Typography>
                            <Typography variant="h5" component="div">
                            {title}
                            </Typography>
                            <Typography sx={{ color: 'text.secondary', mb: 1.5 }}>
                            {type}
                            </Typography>
                            <Typography variant="body2">
                            {assignee}
                            <br />
                            </Typography>
                        </CardContent>
                </CardActionArea>
                {/* <CardActions>
                    <Button size="small">Back</Button>
                    <Button size="small">Forward</Button>
                </CardActions> */}
            </Card>
            // </div>
        // }
        // </Draggable>
    )
}