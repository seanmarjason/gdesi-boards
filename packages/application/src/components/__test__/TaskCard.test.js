import { render } from '@testing-library/react'
import { TaskCard } from '../TaskCard';

jest.mock("next/navigation");
jest.mock("@hello-pangea/dnd");

const card = {
    id: 1,
    title: '',
    type: '',
    assignee: '',
}
 
test('TaskCard renders correctly', () => {
  const { container } = render(<TaskCard card={card} index={1} />)
  expect(container).toMatchSnapshot()
})
