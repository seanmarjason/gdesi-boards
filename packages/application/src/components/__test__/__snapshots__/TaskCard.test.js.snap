// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard renders correctly 1`] = `<div />`;
